### PR TITLE
test: add test cases for signal processing

### PR DESCRIPTION
**Pull request by zhuizhuhaomeng**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -88,6 +88,7 @@ install:
   - git clone https://github.com/openresty/lua-resty-mysql.git ../lua-resty-mysql
   - git clone https://github.com/openresty/stream-lua-nginx-module.git ../stream-lua-nginx-module
   - git clone -b v2.1-agentzh https://github.com/openresty/luajit2.git luajit2
+  - git clone https://github.com/openresty/lua-resty-signal.git ../lua-resty-signal
```
Comment (agentzh) [inline .travis.yml:91]: I'm not feeling comfortable adding this new dependency for the test suite. Can we avoid it? Maybe you can just use `os.execute("kill -HUP " .. pid)` or something like that?

File: .travis.yml [hunk 2]
```diff
@@ -118,6 +119,8 @@ script:
   - ./config no-threads shared enable-ssl3 enable-ssl3-method -g --prefix=$OPENSSL_PREFIX -DPURIFY > build.log 2>&1 || (cat build.log && exit 1)
   - make -j$JOBS > build.log 2>&1 || (cat build.log && exit 1)
   - sudo make PATH=$PATH install_sw > build.log 2>&1 || (cat build.log && exit 1)
+  - cd ../lua-resty-signal/
+  - make
```
Comment (agentzh) [inline .travis.yml:123]: This will require changes in our [EC2 test cluster](https://openresty.org/en/ec2-test-cluster.html) as well. Better don't do it.

**Comment by agentzh** [comment]
Also, the travis ci is failing BTW.

**Comment by agentzh** [comment]
It seems to be duplicate with #1738. Why?

File: .travis.yml (hunk 1 above)
Comment (zhuizhuhaomeng) [reply .travis.yml:91]: OK

File: .travis.yml (hunk 2 above)
Comment (zhuizhuhaomeng) [reply .travis.yml:123]: OK

**Comment by agentzh** [comment]
Merged after fixing some minor grammar issues in the test titles and git log titles.

**Pull request closed without merging**